app/forms/collection_edit_form.rb

```diff
@@ -21,6 +27,8 @@ def self.model_attributes(attrs)
     attrs.dup.each do |key, value|
       if value.is_a? Array
         attrs[key].map! { |v| to_hash(v) }
+      elsif value.nil? || value.to_s.strip.blank?
```

Comment:
It looks like`#compact` will return a new hash while we need to operate on the the attrs hash in place,  So it may not be efficient to loop throughout the k/v in the hash for other operations.